### PR TITLE
[14.0][MIG] project_task_material_stock

### PR DESCRIPTION
**Pull request by omar7r**
Based on https://github.com/OCA/project/pull/725

### CODE REVIEW

File: project_task_material_stock/views/project_task_view.xml [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+                />
+                <button
+                    name="action_done"
+                    type="object"
+                    string="Transfer Materials"
+                    attrs="{'invisible':[('stock_state','!=','assigned')]}"
+                    class="oe_highlight"
+                />
+            </field>
+            <field name="material_ids" position="replace">
```
Comment (dalonsod) [inline project_task_material_stock/views/project_task_view.xml:51]: Since replace is used, maybe this view should have a lower priority

**Review by dalonsod — COMMENTED** [review]
See comment.
Locally tested, LGTM. Not blocking, but is it possible to add `consume_material` field to task type tree and search views? Thanks!

**Comment by dalonsod** [comment]
Hello @omar7r , any news on this?

**Review by BorjaParedes — APPROVED** [review]
Functional review done!

**Review by brendapaniagua — APPROVED** [review]
Functional review! LGTM!

File: project_task_material_stock/models/project_task.py [hunk 2]
```diff
@@ -0,0 +1,280 @@
...
+        }
+        amount_unit = self.product_id.with_context(
+            uom=self.product_uom_id.id
+        ).price_compute("standard_price")[self.product_id.id]
+        amount = amount_unit * self.quantity or 0.0
+        result = round(amount, company_id.currency_id.decimal_places) * -1
+        vals = {"amount": result}
+        if "employee_id" in self.env["account.analytic.line"]._fields:
+            vals["employee_id"] = (
+                self.env["hr.employee"]
```
Comment (sysadminmatmoz) [inline project_task_material_stock/models/project_task.py:234]: the hr dependency should be declared in the manifest

**Comment by pedrobaeza** [comment]
Check `project_stock` module already merged as a more powerful and integrated option.

**Pull request closed without merging**